Submitted By:     VMiklos <vmiklos@frugalware.org>
Date:             2005.03.05
Initial Packages: 2.9.5
Upstream Status:  Not submitted - Frugalware specific
Description:      This patch fixes an annoying bug in makepkg.
                  Instead of removing the documentation, add some automatically.

diff -Naur pacman-2.9.5.orig/scripts/makepkg pacman-2.9.5/scripts/makepkg
--- pacman-2.9.5.orig/scripts/makepkg	2004-12-19 03:09:59.000000000 +0100
+++ pacman-2.9.5/scripts/makepkg	2005-02-03 14:58:50.048936144 +0100
@@ -560,10 +560,36 @@
 	exit 2
 fi
 
-# remove info/doc files
+# documentation
+msg "Preparing package documentation..."
 cd $startdir
-rm -rf pkg/usr/info pkg/usr/share/info
-rm -rf pkg/usr/doc pkg/usr/share/doc
+if [ -d pkg/usr/doc ]; then
+	mkdir -p pkg/usr/share/doc
+	cp -a pkg/usr/doc/* pkg/usr/share/doc/
+	rm -rf pkg/usr/doc
+fi
+mkdir -p pkg/usr/share/doc/$pkgname-$pkgver
+for i in `find src -maxdepth 2 -name ANNOUNCE -o -name AUTHORS -o -name *BUGS* -o -name CHANGES -o -name CONFIGURATION -o -name *COPYING* -o -name *COPYRIGHT* -o -name CREDITS -o -name ChangeLog -o -name Changelog -o -name CHANGELOG -o -name CONTRIBUTORS -o -name *FAQ* -o -name FEATURES -o -name FILES -o -name HACKING -o -name History -o -name HISTORY -o -name INSTALL* -o -name LICENSE -o -name LSM -o -name MANIFEST -o -name NEWS -o -name *README* -o -name *Readme* -o -name SITES -o -name *RELEASE* -o -name RELNOTES -o -name THANKS -o -name TIPS -o -name TODO -o -name VERSION -o -name CONFIGURATION* -o -name GPLLicense`
+do
+	cp -a $i pkg/usr/share/doc/$pkgname-$pkgver/
+done
+
+# remove /usr/share/doc/$pkgname-$pkgver directory if empty
+if [ -z "`ls -1 pkg/usr/share/doc/$pkgname-$pkgver`" ]; then
+       rmdir pkg/usr/share/doc/$pkgname-$pkgver
+fi
+
+# remove /usr/share/doc/ directory if empty
+if [ -z "`ls -1 pkg/usr/share/doc`" ]; then
+       rmdir pkg/usr/share/doc
+fi
+
+# move /usr/share/info files to /usr/info
+if [ -d pkg/usr/share/info ]; then
+	mkdir -p pkg/usr/info
+	cp -a pkg/usr/share/info/* pkg/usr/info/
+	rm -rf pkg/usr/share/info
+fi
 
 # move /usr/share/man files to /usr/man
 if [ -d pkg/usr/share/man ]; then
--- pacman-2.9.5.orig/doc/makepkg.8.in	2005-08-07 23:50:35.000000000 +0200
+++ pacman-2.9.5/doc/makepkg.8.in	2005-08-07 23:53:48.000000000 +0200
@@ -20,10 +20,11 @@
 than building this package).  The build script should be called FrugalBuild
 and it should bear resemblance to the example below.
 
-\fBNOTE:\fP If you have a local copy of the Arch Build System (ABS) tree
-on your computer, you can copy the FrugalBuild.proto file to your new package
-build directory and edit it from there.  To acquire/sync the ABS tree, use
-the \fBabs\fP script included with pacman/makepkg.
+\fBNOTE:\fP If you have a local copy of the Frugalware Source Tree (FST)
+on your computer, you can copy the source/apps/cabextract/FrugalBuild
+file to your new package
+build directory and edit it from there.  To acquire/sync the FST, use
+the \fBrepoman upd\fP command provided by the pacman-tools package.
 
 .TP
 .TP
@@ -57,7 +58,7 @@
 
 As you can see, the setup is fairly simple.  The first three lines define
 the package name and version info.  They also define the final package name
-which will be of the form \fI$pkgname-$pkgver-$pkgrel.fpm\fP.  The fourth
+which will be of the form \fI$pkgname-$pkgver-$pkgrel-$arch.fpm\fP.  The fourth
 line provides a brief description of the package.  These four lines should
 be present in every FrugalBuild script.
 
@@ -104,14 +105,13 @@
 "prefix" only when "DESTDIR" is unavailable.
 
 Once the package is successfully installed into the package root, \fImakepkg\fP
-will remove some directories (as per Arch Linux package guidelines; if you use
-this elsewhere, feel free to change it) like /usr/doc and /usr/info.  It will
+will prepare some documentation. It will
 then strip debugging info from libraries and binaries and generate a meta-info
 file.  Finally, it will compress everything into a .fpm file and leave it
 in the directory you ran \fBmakepkg\fP from.
 
 At this point you should have a package file in the current directory, named
-something like name-version-release.fpm.  Done!
+something like name-version-release-arch.fpm.  Done!
 
 .SH Install/Upgrade/Remove Scripting
 Pacman has the ability to store and execute a package-specific script when it
@@ -229,7 +229,7 @@
 
 .TP
 .B pkgrel
-This is the release number specific to Arch Linux packages.
+This is the release number specific to Frugalware Linux packages.
 
 .TP
 .B pkgdesc
@@ -343,7 +343,7 @@
 process if all of the dependencies aren't installed.
 .TP
 .B "\-f, \-\-force"
-\fBmakepkg\fP will not build a package if a \fIpkgname-pkgver-pkgrel.fpm\fP
+\fBmakepkg\fP will not build a package if a \fIpkgname-pkgver-pkgrel-arch.fpm\fP
 file already exists in the build directory.  You can override this behaviour with
 the \fB--force\fP switch.
 .TP
@@ -399,9 +399,10 @@
 .SH SEE ALSO
 \fBpacman\fP is the package manager that uses packages built by makepkg.
 
-See the Arch Linux Documentation for package-building guidelines if you wish
-to contribute packages to the Arch Linux project.
+See the Frugalware Linux Documentation for package-building guidelines if you
+wish to contribute packages to the Frugalware Linux project.
 .SH AUTHOR
 .nf
 Judd Vinet <jvinet@zeroflux.org>
+and the Frugalware developers <frugalware-devel@frugalware.org>
 .fi
diff -Naur pacman-2.9.6.orig/src/pacman.c pacman-2.9.6/src/pacman.c
--- pacman-2.9.6.orig/src/pacman.c	2005-04-25 20:19:24.000000000 +0200
+++ pacman-2.9.6/src/pacman.c	2005-08-27 00:49:05.000000000 +0200
@@ -3827,8 +3827,9 @@
 	printf("\n");
 	printf(" .--.                  Pacman v%s\n", PACVER);
 	printf("/ _.-' .-.  .-.  .-.   Copyright (C) 2002-2005 Judd Vinet <jvinet@zeroflux.org>\n");
-	printf("\\  '-. '-'  '-'  '-'  \n");
-	printf(" '--'                  This program may be freely redistributed under\n");
+	printf("\\  '-. '-'  '-'  '-'   & Frugalware developers <frugalware-devel@frugalware.org>\n");
+	printf(" '--'                  \n");
+	printf("                       This program may be freely redistributed under\n");
 	printf("                       the terms of the GNU General Public License\n\n");
 }
 
